packages/block-library/src/template-part/edit/inner-blocks.js

```diff
@@ -4,16 +4,25 @@
 import { useEntityBlockEditor } from '@wordpress/core-data';
 import { InnerBlocks } from '@wordpress/block-editor';
 
-export default function TemplatePartInnerBlocks() {
+export default function TemplatePartInnerBlocks( {
+	postId: id,
+	hasInnerBlocks,
+} ) {
 	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
-		'wp_template_part'
+		'wp_template_part',
+		{ id }
 	);
 	return (
 		<InnerBlocks
 			value={ blocks }
 			onInput={ onInput }
 			onChange={ onChange }
+			renderAppender={
+				hasInnerBlocks
+					? undefined
+					: () => <InnerBlocks.ButtonBlockAppender />
```

Comment:
Should this function be defined as opposed to anonymous?

I think anonymous functions passed into component props cause unnecessary re-renders as `() => {}  !==  () => {}` and the component always views that prop as having changed?

Reply:
Only if the component is "Pure", `InnerBlocks` isn't last time I checked. Still, it doesn' make it that much more complex in this instance, so I've updated it.